connector/ldap/ldap.go

```diff
@@ -188,12 +188,12 @@ func parseScope(s string) (int, bool) {
 // Function exists here to allow backward compatibility between old and new
 // group to user matching implementations.
 // See "Config.GroupSearch.UserMatchers" comments for the details
-func userMatchers(c *Config, logger log.Logger) []UserMatcher {
+func userMatchers(c *Config, logger *slog.Logger) []UserMatcher {
 	if len(c.GroupSearch.UserMatchers) > 0 && c.GroupSearch.UserMatchers[0].UserAttr != "" {
 		return c.GroupSearch.UserMatchers
 	}
 
-	log.Deprecated(logger, `LDAP: use groupSearch.userMatchers option instead of "userAttr/groupAttr" fields.`)
+	slog.Warn(`use "groupSearch.userMatchers" option instead of "userAttr/groupAttr" fields`, "type", "deprecated", "connector", "ldap")
```

Comment:
Should we use the logger here? Otherwise the logline will not be formatted as JSON. 

BTW what is type? Maybe we can use a boolean label like `"deprecated": true` similar to Kubernrtes?

Reply:
I was a bit unsure about it since the old deprecation messages logged through a different logger.
I was thinking of the type of log message when I originally went with type=deprecated.

I've changed it to use the component's logger and type=deprecated -> deprecated=true (plus 2 other places where deprecations were logged).